assets/js/modules/analytics/datastore/settings.js

```diff
@@ -126,6 +129,10 @@ export function validateCanSubmitChanges( select ) {
 	invariant( isValidPropertySelection( getPropertyID() ), INVARIANT_INVALID_PROPERTY_SELECTION );
 	invariant( isValidProfileSelection( getProfileID() ), INVARIANT_INVALID_PROFILE_SELECTION );
 
+	if ( getAdsConversionID() ) {
+		invariant( isValidAdsConversionID( getAdsConversionID() ), INVARIANT_INVALID_CONVERSION_ID );
+	}
```

Comment:
This could result in the user being blocked if they were to get an invalid value in the setting (either saved or just the current unsaved value) if the field is then hidden. I've updated the conditional display to be handled by the component itself in the edit view now so that it's only hidden if it has a valid value.